### PR TITLE
Fix Bad Auto 4 (with a bonus)

### PR DESCRIPTION
**Pull request by amirroth**
This is the fourth and final chunk of the pass to remove bad usages of plain (non-ref) `auto` from the code. Note: the second chunk has not been made into a PR yet and will probably have to be thrown away and redone after this release.

Bonus: with all the compilers we are using now doing the `BITF` trick on -O2 (or /O2), there is no need for us to roll this optimization by hand.

### CODE REVIEW

File: src/EnergyPlus/BITF.hh
Comment (amirroth) [inline src/EnergyPlus/BITF.hh:1]: Buh-bye!

File: src/EnergyPlus/OutdoorAirUnit.cc [hunk 1]
```diff
@@ -233,17 +233,9 @@ namespace OutdoorAirUnit {
         using OutAirNodeManager::CheckAndAddAirNodeNumber;
         using WaterCoils::GetCoilWaterInletNode;
         using WaterCoils::GetWaterCoilIndex;
-        auto &GetWCoilInletNode(WaterCoils::GetCoilInletNode);
-        auto &GetWCoilOutletNode(WaterCoils::GetCoilOutletNode);
         using HeatingCoils::GetCoilInletNode;
         using HeatingCoils::GetCoilOutletNode;
         using WaterCoils::GetCoilWaterOutletNode;
-        auto &GetHeatingCoilIndex(HeatingCoils::GetCoilIndex);
```
Comment (amirroth) [inline src/EnergyPlus/OutdoorAirUnit.cc:241]: This is an abomination and I will have none of it.

File: src/EnergyPlus/OutdoorAirUnit.cc [hunk 2]
```diff
@@ -530,7 +522,7 @@ namespace OutdoorAirUnit {
             if (!lAlphaBlanks(9)) {
                 constexpr std::array<std::string_view, static_cast<int>(OAUnitCtrlType::Num)> ctrlTypeNamesUC = {
                     "NEUTRALCONTROL", "INVALID-UNCONDITIONED", "TEMPERATURECONTROL"};
-                auto tmpCtrlType = static_cast<OAUnitCtrlType>(getEnumerationValue(ctrlTypeNamesUC, state.dataIPShortCut->cAlphaArgs(9)));
+                OAUnitCtrlType tmpCtrlType = static_cast<OAUnitCtrlType>(getEnumerationValue(ctrlTypeNamesUC, state.dataIPShortCut->cAlphaArgs(9)));
```
Comment (amirroth) [inline src/EnergyPlus/OutdoorAirUnit.cc:525]: The usual.

File: src/EnergyPlus/OutdoorAirUnit.cc [hunk 3]
```diff
@@ -1215,7 +1207,7 @@ namespace OutdoorAirUnit {
 
         // Do the one time initializations
 
-        auto &OutAirUnit(state.dataOutdoorAirUnit->OutAirUnit);
+        auto &OutAirUnit = state.dataOutdoorAirUnit->OutAirUnit;
```
Comment (amirroth) [inline src/EnergyPlus/OutdoorAirUnit.cc:1210]: Let's not use constructor initialization notation when simple scalar assignment will do. Also, it messes up my `grep` scripts.

File: src/EnergyPlus/PluginManager.cc [hunk 4]
```diff
@@ -609,7 +609,7 @@ PluginManager::PluginManager(EnergyPlusData &state) : eplusRunningViaPythonAPI(s
             auto const &fields = instance.value();
             std::string const &thisObjectName = instance.key();
             state.dataInputProcessing->inputProcessor->markObjectAsUsed(sGlobals, thisObjectName);
-            auto const vars = fields.at("global_py_vars");
+            auto const &vars = fields.at("global_py_vars");
```
Comment (amirroth) [inline src/EnergyPlus/PluginManager.cc:612]: Reminder, `x.at()` returns a reference to a persistent object with the field map. Without `&`, you are making a local copy of that object, which is often not what you want.

File: src/EnergyPlus/Pumps.cc [hunk 5]
```diff
@@ -2145,7 +2145,7 @@ void SizePump(EnergyPlusData &state, int const PumpNum)
     }
 
     // Note that autocalculation of power is based on nominal volume flow, regardless of whether the flow was
-    //  auto sized or manually sized.  Thus, this must go after the flow sizing block above.
+    //  auto-sized or manually sized.  Thus, this must go after the flow sizing block above.
```
Comment (amirroth) [inline src/EnergyPlus/Pumps.cc:2148]: So that my `grep` scripts don't think this is an `auto`

File: src/EnergyPlus/SZVAVModel.cc [hunk 6]
```diff
@@ -338,7 +338,7 @@ namespace SZVAVModel {
                         // this is different from the PTUnit and UnitarySys routines in this module
                         // find the water flow rate that meets the min load at region 1/2 bounday
                         if (SZVAVModel.HCoilType_Num == FanCoilUnits::HCoil::Water || !HeatingLoad) {
-                            auto f =
+                            auto f = // (AUTO_OK_LAMBDA)
```
Comment (amirroth) [inline src/EnergyPlus/SZVAVModel.cc:341]: `grep` script cannot tell this is a lambda if the `[` is not on the same line as the `auto` (`grep` works one text line at a time).

File: src/EnergyPlus/ScheduleManager.cc [hunk 7]
```diff
@@ -485,7 +485,7 @@ namespace ScheduleManager {
                     schedule_file_shading_result = it.first;
                 } else if (FileSystem::is_all_json_type(ext)) {
                     auto schedule_data = FileSystem::readJSON(state.files.TempFullFilePath.filePath);
-                    auto it =
+                    auto it = // (AUTO_OK_ITER)
```
Comment (amirroth) [inline src/EnergyPlus/ScheduleManager.cc:488]: Same thing, `grep` cannot tell this is an iterator if `.find`, `.begin`, `.end`, `.next`, or `.emplace` is not on the same line.

File: src/EnergyPlus/ScheduleManager.cc [hunk 8]
```diff
@@ -1927,8 +1927,8 @@ namespace ScheduleManager {
 
         if (NumCommaFileShading != 0) {
             auto const &values_json = schedule_file_shading_result->second["values"];
-            auto const headers = schedule_file_shading_result->second["header"].get<std::vector<std::string>>();
-            auto const headers_set = schedule_file_shading_result->second["header"].get<std::set<std::string>>();
+            auto const headers = schedule_file_shading_result->second["header"].get<std::vector<std::string>>(); // (AUTO_OK_OBJ)
```
Comment (amirroth) [inline src/EnergyPlus/ScheduleManager.cc]: I can probably add a special case to my script to capture this case. I will do that later.

File: src/EnergyPlus/TARCOGArgs.cc [hunk 9]
```diff
@@ -425,10 +425,13 @@ int ArgCheck(EnergyPlusData &state,
             return ArgCheck;
         }
 
-        if (BITF_TEST_NONE(BITF(LayerType(i)),
-                           BITF(TARCOGLayerType::SPECULAR) | BITF(TARCOGLayerType::WOVSHADE) | BITF(TARCOGLayerType::VENETBLIND_HORIZ) |
-                               BITF(TARCOGLayerType::PERFORATED) | BITF(TARCOGLayerType::DIFFSHADE) | BITF(TARCOGLayerType::BSDF) |
-                               BITF(TARCOGLayerType::VENETBLIND_VERT)))
+        if (LayerType(i) != TARCOGLayerType::SPECULAR &&
```
Comment (amirroth) [inline src/EnergyPlus/TARCOGArgs.cc]: A bit more readable and now just as fast.

**Review by Myoldmopar — APPROVED** [review]
This is so awesome, and I'm so glad to see BITF go away.  It served its purpose, but so glad we don't need it anymore.  :+1:

**Comment by Myoldmopar** [comment]
Minor conflict here, I'll resolve and apply format.

**Comment by Myoldmopar** [comment]
OK, everything is happy locally.  I'll still let CI make at least one pass, but I anticipate this will merge later this afternoon.  I'll probably tag RC1 just for fun after that merges too.  Any new PRs should target 23.2+.

**Comment by amirroth** [comment]
> Minor conflict here, I'll resolve and apply format.

This happened because I didn't apply `clang-format` locally. Sorry.

**Pull request merged**